stdlib/sql/to.go

```diff
@@ -207,38 +222,95 @@ func (t *ToSQLTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
 
+type translationFunc func(f flux.ColType, colname string) (string, error)
+
+func correctBatchSize(batchSize, numberCols int) int {
+	/*
+		BatchSize for the DB is the number of parameters that can be queued within each call to Exec.
+		As each row you send has a parameter count equal to the number of columns (i.e. the number of "?" used in the insert statement), and some DBs
+		have a default limit on the number of parameters which can be queued before calling Exec; SQLite, for example, has a default of 999 (can only be changed
+		at compile time).
+
+		So if the row width is 10 columns, the maximum Batchsize would be:
+		(999 - row_width) / row_width = 89 rows.
+
+		Sending more would result in the call to Exec returning a "too many SQL variables" error, and the transaction would be rolled-back / aborted
+	*/
+
+	if batchSize < numberCols {
```

Comment:
@nathanielc can `numberCols` be greater than 999 in flux?

Reply:
Good point - if it can, then we would be inserting rows which exceed the number of Parma’s that the DB will accept in an insert. Likely this would also impact many other dbs. Perhaps returning an error so user can ‘do something’ about limiting the width of the data?